src/components/views/dialogs/SessionRestoreErrorDialog.js

```diff
@@ -31,63 +31,73 @@ export default React.createClass({
         onFinished: PropTypes.func.isRequired,
     },
 
-    componentDidMount: function() {
-        if (this.refs.bugreportLink) {
-            this.refs.bugreportLink.focus();
-        }
-    },
-
     _sendBugReport: function() {
         const BugReportDialog = sdk.getComponent("dialogs.BugReportDialog");
         Modal.createTrackedDialog('Session Restore Error', 'Send Bug Report Dialog', BugReportDialog, {});
     },
 
-    _onContinueClick: function() {
-        this.props.onFinished(true);
+    _onClearStorageClick: function() {
+        const QuestionDialog = sdk.getComponent("dialogs.QuestionDialog");
+        Modal.createTrackedDialog('Session Restore Confirm Logout', '', QuestionDialog, {
+            title: _t("Sign out"),
+            description:
+                <div>{ _t("Log out and remove encryption keys?") }</div>,
+            button: _t("Sign out"),
+            danger: true,
+            onFinished: this.props.onFinished,
+        });
     },
 
-    _onCancelClick: function() {
-        this.props.onFinished(false);
+    _onRefreshClick: function() {
+        // Is this likely to help? Probably not, but giving only one button
+        // that clears your storage seems awful.
+        window.location.reload(true);
     },
 
     render: function() {
         const BaseDialog = sdk.getComponent('views.dialogs.BaseDialog');
         const DialogButtons = sdk.getComponent('views.elements.DialogButtons');
-        let bugreport;
 
+        let dialogButtons;
         if (SdkConfig.get().bug_report_endpoint_url) {
-            bugreport = (
-                <p>
-                { _t(
-                    "Otherwise, <a>click here</a> to send a bug report.",
-                    {},
-                    { 'a': (sub) => <a ref="bugreportLink" onClick={this._sendBugReport}
-                    key="bugreport" href='#'>{ sub }</a> },
-                ) }
-                </p>
-            );
+            dialogButtons = <DialogButtons primaryButton={_t("Send Logs")}
+                onPrimaryButtonClick={this._sendBugReport}
+                focus={true}
+                hasCancel={false}
+            >
+                <button onClick={this._onClearStorageClick} className="danger">
+                    { _t("Clear Storage and Sign Out") }
+                </button>
+            </DialogButtons>;
+        } else {
+            dialogButtons = <DialogButtons primaryButton={_t("Refresh")}
+                onPrimaryButtonClick={this._onRefreshClick}
+                focus={true}
+                hasCancel={false}
+            >
+                <button onClick={this._onClearStorageClick} className="danger">
+                    { _t("Clear Storage and Sign Out") }
```

Comment:
This looks identical to the `if` block apart from the text "Refresh". 

OH. I see what this is doing now. Maybe factor out the "Clear Storage..." button and refer to it twice.

Reply:
mm, that is probably better. done